### PR TITLE
[DRAFT] Introduce circle-interpreter-cffi-test

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces a library for C-Python interaction.

Draft: #14352
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: compiler/circle-interpreter-cffi-test/infer.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+parser = argparse.ArgumentParser()
+parser.add_argument('--lib_path', type=str, required=True)
+parser.add_argument('--test_list', type=str, required=True)
+parser.add_argument('--artifact_dir', type=str, required=True)
+args = parser.parse_args()
+
+with open(args.test_list) as f:
+    contents = [line.rstrip() for line in f]
+comment_removed = [line for line in contents if not line.startswith('#')]
+newline_removed = [line for line in comment_removed if line.startswith('eval(')]
```
Comment (jinevening) [inline compiler/circle-interpreter-cffi-test/infer.py:26]: comment_removed seems redundant.

```suggestion
contents = [line for line in contents if line.startswith('eval(')]
```

File: compiler/circle-interpreter-cffi-test/infer.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    for line in newline_removed
+]
+input_data = [
+    Path(args.artifact_dir) / f'{extract_test_args(line)}.opt/metadata/tc/input.h5'
+    for line in newline_removed
+]
+expected_output_data = [
+    Path(args.artifact_dir) / f'{extract_test_args(line)}.opt/metadata/tc/expected.h5'
+    for line in newline_removed
+]
```
Comment (jinevening) [inline compiler/circle-interpreter-cffi-test/infer.py:38]: Better to call extract_trest_args once.

File: compiler/circle-interpreter-cffi-test/infer.py (hunk 1 above)
Comment (mhs4670go) [reply compiler/circle-interpreter-cffi-test/infer.py:26]: I added this in advance to make it possible to have comment in `test.lst`.

Comment (jinevening) [reply compiler/circle-interpreter-cffi-test/infer.py:26]: comment would be automatically removed with `if line.startswith('eval(')`.

Comment (mhs4670go) [reply compiler/circle-interpreter-cffi-test/infer.py:26]: Oh, I mean in-line comment like this.
```
eval(~~~) # note this test is ~
```

**Pull request closed without merging**